tests/trestle/core/commands/validate_test.py

```diff
@@ -296,7 +303,107 @@ def test_validate_catalog_params(sample_catalog_rich_controls: Catalog) -> None:
+    assert validator.model_is_valid(sample_catalog_rich_controls, True, None)
     param_0_id = sample_catalog_rich_controls.groups[0].controls[0].params[0].id
     sample_catalog_rich_controls.groups[0].controls[0].params[1].id = param_0_id
-    assert not validator.model_is_valid(sample_catalog_rich_controls, False)
+    assert not validator.model_is_valid(sample_catalog_rich_controls, False, None)
+
+
+def test_rule_param_values_validator_happy(tmp_trestle_dir: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
+    """Test validation of rules in SSP."""
+    args = argparse.Namespace(mode=const.VAL_MODE_RULES)
+    validator: Validator = validator_factory.get(args)
+    prof_name = 'comp_prof'
+    ssp_name = 'new_ssp'
+
+    gen_args, _ = setup_for_ssp(tmp_trestle_dir, prof_name, ssp_name)
+    test_utils.gen_and_assemble_first_ssp(prof_name, ssp_name, gen_args, tmp_trestle_dir, monkeypatch)
+
+    new_ssp, _ = ModelUtils.load_model_for_class(tmp_trestle_dir, ssp_name, ossp.SystemSecurityPlan)
+    imp_reqs = new_ssp.control_implementation.implemented_requirements
+    imp_req = next((i_req for i_req in imp_reqs if i_req.control_id == 'ac-3'), None)
+    imp_req.by_components[0].set_parameters[1].values[0] = 'shared_param_1_aa_opt_1'
+    ModelUtils.save_top_level_model(new_ssp, tmp_trestle_dir, ssp_name, FileContentType.JSON)
+
+    assert validator.model_is_valid(new_ssp, True, tmp_trestle_dir)
+
+
+def test_validate_ssp_with_no_profile(tmp_trestle_dir: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
+    """Test validation of rules in SSP."""
+    args = argparse.Namespace(mode=const.VAL_MODE_RULES)
+    validator: Validator = validator_factory.get(args)
+    prof_name = 'comp_prof'
+    ssp_name = 'new_ssp'
+
+    gen_args, _ = setup_for_ssp(tmp_trestle_dir, prof_name, ssp_name)
+    test_utils.gen_and_assemble_first_ssp(prof_name, ssp_name, gen_args, tmp_trestle_dir, monkeypatch)
+
+    new_ssp, _ = ModelUtils.load_model_for_class(tmp_trestle_dir, ssp_name, ossp.SystemSecurityPlan)
+
+    original_href = new_ssp.import_profile.href
+    # sets href for given profile to empty to test if no given profile has been given to an ssp
+    new_ssp.import_profile.href = ''
+    ModelUtils.save_top_level_model(new_ssp, tmp_trestle_dir, ssp_name, FileContentType.JSON)
+
+    assert not validator.model_is_valid(new_ssp, True, tmp_trestle_dir)
+
+    validate_command = f'trestle validate -t system-security-plan -n {ssp_name}'
+    test_utils.execute_command_and_assert(validate_command, 4, monkeypatch)
+
+    new_ssp.import_profile.href = original_href
+    ModelUtils.save_top_level_model(new_ssp, tmp_trestle_dir, ssp_name, FileContentType.JSON)
+
+
+def test_rule_param_values_validator_unhappy(tmp_trestle_dir: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
+    """Test validation of rules in SSP."""
+    args = argparse.Namespace(mode=const.VAL_MODE_RULES)
+    validator: Validator = validator_factory.get(args)
```

Comment:
pls move this test up to be next to the happy one

Reply:
done